app/workers/pubsubhubbub/distribution_worker.rb

```diff
@@ -12,10 +12,10 @@ def perform(stream_entry_id)
 
     account = stream_entry.account
     payload = AtomSerializer.render(AtomSerializer.new.feed(account, [stream_entry]))
-    domains = account.followers_domains
+    domains = account.followers_subscription_domains if stream_entry.status&.hidden?
 
     Subscription.where(account: account).active.select('id, callback_url').find_each do |subscription|
-      next unless domains.include?(Addressable::URI.parse(subscription.callback_url).host)
+      next if stream_entry.status&.hidden? && !domains.include?(Addressable::URI.parse(subscription.callback_url).normalize.host)
```

Comment:
I'm afraid I'm out of ideas to make this efficient. I guess we could add the database field right now, without doing the full migration, but with everything to compute the web_domain whenever the account details are updated. Then, we can fetch distinct non-null web_domain from the followers, and do the big array thing for the remaining remote followers.